### PR TITLE
Bill startup plan overages

### LINKED ISSUE #92: Improve logic for startup billing plans
Currently, the logic for startup plans is handled on `posthog-cloud` (free pricing, and expiring the plan after 365 days). This however can be improved and the billing plan logic handled on Stripe instead of on our end. The general idea is we would have either a) a startup plan that only lasts for a year and is free for the year and they would then transition to a standard plan.

### Benefits
- All billing logic centralized on Stripe, less error prone / prevent complexity in multiple systems.
- Stripe can be the source of truth for all this information. Currently we don't have startup plan information on Stripe.
- We can do complex and custom billing agreements without requiring code changes.
- Stripe will automatically handle the transition to a regular plan after the startup plan expires. Currently, when a startup runs out their plan, they (or we) will need to start a new billing agreement.

### PR DESCRIPTION
**Pull request by paolodamico**
Closes #92. From recent [user feedback](https://posthog.slack.com/archives/C01LNUMT022/p1636375893001300), currently if a user goes over the 20M events on the startup plan (or 5M for flat 5), they get the exceeded warning alert with no remedy.

This changes the way startup plan (regular and flat 5) work:
- We will now create a full subscription on Stripe which ends after 1 year. The subscriptions has 20M/5M events/month for free respectively, after which overages are billed at regular rates.
- Subscriptions end automatically after 1 year.
- Subscriptions are already set and good to go on PostHog Cloud & Stripe.

**Next steps**
- Smooth transition from startup plan to standard plan.

### CODE REVIEW

**Comment by paolodamico** [comment]
And here's the code **proposal** we can run ad-hoc, to update the currently active startup plan subscribers, **please review too:**

```python
from multi_tenancy.models import OrganizationBilling, Plan
from django.utils import timezone
from multi_tenancy.stripe import create_subscription


for org_bill in OrganizationBilling.objects.filter(plan__key__in=["startup", "flat-5"], billing_period_ends__gte=timezone.now()):
    print(f"Updating subscription for: {org_bill.organization.name} ({org_bill.organization.id}), current end date: {org_bill.billing_period_ends}")
    
    subscription = create_subscription(
        price_id=org_bill.plan.price_id, customer_id=org_bill.stripe_customer_id, cancel_at=org_bill.billing_period_ends,
    )
    org_bill.stripe_subscription_item_id = subscription["subscription_item_id"]
    org_bill.stripe_subscription_id = subscription["subscription_id"]
    org_bill.should_setup_billing = False
    org_bill.billing_period_ends = None

    org_bill.save()
```

**Comment by kpthatsme** [comment]
> ```python
> stripe_subscription_id
> ```

@paolodamico Should we add a check to this script to not pull Org Bills that already have a `stripe_subscription_id` set?

File: multi_tenancy/models.py [hunk 1]
```diff
@@ -160,14 +159,20 @@ def handle_post_card_validation(self) -> "OrganizationBilling":
         """
         Handles logic after a card has been validated.
         """
+        subscription = None
         if self.plan.key == "startup" or self.plan.key == "flat-5":
-            self.billing_period_ends = timezone.now() + datetime.timedelta(days=365)
-            self.should_setup_billing = False
+            a_year_from_now = timezone.now() + datetime.timedelta(days=365)
+            subscription = create_subscription(
+                price_id=self.plan.price_id, customer_id=self.stripe_customer_id, cancel_at=a_year_from_now
```
Comment (kpthatsme) [inline multi_tenancy/models.py:166]: Just for the sake of my own knowledge-

Is it safe to assume we've created a Stripe customer for all of these users already? So in the old flow we were already creating customers and only forgoing the subscriptions?

**Comment by paolodamico** [comment]
We shouldn't have anyone in that case @kpthatsme, but better safe than sorry, updated the script.

File: multi_tenancy/models.py (hunk 1 above)
Comment (paolodamico) [reply multi_tenancy/models.py:166]: Yes that's a pretty safe assumption because if they registered a card, they need to have a `customer_id`

**Comment by paolodamico** [comment]
Ping here for a review

**Comment by paolodamico** [comment]
Ping @kpthatsme do we want to move forward with this or should we scrap it?

**Review by kpthatsme — APPROVED** [review]
sorry about that @paolodamico  - lgtm!

**Pull request merged**